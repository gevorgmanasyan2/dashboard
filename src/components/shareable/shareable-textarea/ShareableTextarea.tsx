import React, { FC } from 'react'
import { Input } from 'antd'
import { SharableProps } from '../models'
import './shareableTextarea.scss'

const ShareableTextarea: FC<SharableProps> = ({errorsMessage,className,onBlur, label, value, name, onChange,onKeyDown,autoSize }) => {
  const { TextArea } = Input
  return (
    <div className="textarea__container">
      <label htmlFor="">{label}</label>
      <TextArea
        rows={4}
        className={`textarea__container--textarea ${className}`}
        value={value}
        onBlur={onBlur}
        name={name}
        onChange={onChange}
        onKeyDown={onKeyDown}
        autoSize={autoSize}
      />
      {errorsMessage && <div className='textarea__container--error'>{errorsMessage}</div>}
    </div>
  )
}

export default ShareableTextarea
