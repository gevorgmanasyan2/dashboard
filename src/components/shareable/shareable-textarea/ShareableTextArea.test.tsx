/* eslint-disable @typescript-eslint/no-empty-function */
/* eslint-disable no-undef */
import React from 'react'
import { shallow } from 'enzyme'
import ShareableTextarea from './ShareableTextarea'

it('should render ShareableSelect component', () => {
  const mockItem = {
    label: 'Description',
    onChange: () => {},
    onBlur: () => {},
    errorsMessage: 'error',
    className: 'text',
    value: 'test',
    name: 'test',
  }
  expect(shallow(<ShareableTextarea {...mockItem} />)).toMatchSnapshot()
})
