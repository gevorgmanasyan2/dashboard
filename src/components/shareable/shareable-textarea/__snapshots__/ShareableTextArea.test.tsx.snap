// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render ShareableSelect component 1`] = `
<div
  className="textarea__container"
>
  <label
    htmlFor=""
  >
    Description
  </label>
  <TextArea
    className="textarea__container--textarea text"
    name="test"
    onBlur={[Function]}
    onChange={[Function]}
    rows={4}
    value="test"
  />
  <div
    className="textarea__container--error"
  >
    error
  </div>
</div>
`;
